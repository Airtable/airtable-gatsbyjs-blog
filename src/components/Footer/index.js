import React from 'react';
import styled, { css } from 'react-emotion';
import Link from 'gatsby-link';
import PropTypes from 'prop-types';
import { Box, Flex } from '../Layout';
import feather from '../../utils/feather';
import media from '../../utils/media';

const footerStyle = css`
  overflow: hidden;
  text-align: center;

  & img {
    display: block;
    margin: 0;
  }
  & p {
    color: rgba(255, 255, 255, 0.5);
  }
`;

const ulStyle = css`
  list-style-type: none;
  margin: 0;
  margin-bottom: 1.45rem;
  text-transform: uppercase;
  padding: 0;
  & a {
    cursor: pointer;
    opacity: 0.5;
    transition: opacity 0.15s ease-in;
    transition: color 0.15s ease-in;

    &:hover {
      text-decoration: none;
      box-shadow: none;
      opacity: 1;
      transition: opacity 0.15s ease-in;
    }
  }
  & li {
    margin: 0;
  }
`;

const Wrapper = styled.div`
  width: 100%;
  margin: 0;
  & p {
    margin: 0;
    margin-bottom: 1.45rem;
  }
`;

const socialList = css`
  display: flex;
  justify-content: space-between;
  list-style-type: none;
  margin: 0;
  padding: 0;

  ${media.large`
    width: 50%;
    margin-left: auto;
  `};
`;

const svgStyles = css`
  opacity: 0.5;
  transition: opacity 0.15s ease-in;
  transition: color 0.15s ease-in;

  &:hover {
    text-decoration: none;
    box-shadow: none;
    opacity: 1;
    transition: opacity 0.15s ease-in;
  }
`;

const Footer = props => (
  <div className={footerStyle}>
    <Box p={[3, 3, 4]} align="left" color="white">
      <Flex wrap={['wrap', 'wrap', 'nowrap']} justify="space-between">
        <Box width={[1, 1, 1 / 3]}>
          <Wrapper>
            <div>
              <p>{props.name}</p>
            </div>
          </Wrapper>
        </Box>
        <Box width={[1, 1, 1 / 3]}>
          <Wrapper>
            <ul className={ulStyle}>
              <li>
                <Link to="/">Home</Link>
              </li>
              <li>
                <Link to="/privacy">Privacy Policy</Link>
              </li>
              <li>
                <Link to="/terms">Terms of Service</Link>
              </li>
            </ul>
          </Wrapper>
        </Box>
        <Box width={[1, 1, 1 / 3]}>
          <Wrapper>
            <ul className={socialList}>
              <li>
                <a
                  title="Link to our Facebook page"
                  href={props.facebook}
                  target="_blank"
                  rel="noopener"
                >
                  {feather('facebook', ['30', '30'], svgStyles)}
                </a>
              </li>
              <li>
                <a
                  title="Link to our Twitter account"
                  href={props.twitter}
                  target="_blank"
                  rel="noopener"
                >
                  {feather('twitter', ['30', '30'], svgStyles)}
                </a>
              </li>
              <li>
                <a
                  title="Link to out Instragram account"
                  href={props.instagram}
                  target="_blank"
                  rel="noopener"
                >
                  {feather('instagram', ['30', '30'], svgStyles)}
                </a>
              </li>
              <li>
                <a title="Our E-Mail address" href={`mailto:${props.email}`}>
                  {feather('mail', ['30', '30'], svgStyles)}
                </a>
              </li>
            </ul>
          </Wrapper>
        </Box>
      </Flex>
    </Box>
    <Box px={[3, 3, 1]} align="center">
      <p>{`Copyright Â© 2017 ${props.name}. All rights reserved.`}</p>
    </Box>
  </div>
);

Footer.propTypes = {
  name: PropTypes.string,
  facebook: PropTypes.string,
  twitter: PropTypes.string,
  instagram: PropTypes.string,
  email: PropTypes.string
};

Footer.defaultProps = {
  name: 'Gatbsythemes.com starter',
  facebook: 'https://facebook.com',
  twitter: 'https://twitter.com',
  instagram: 'https://instagram',
  email: ''
};

export default Footer;
