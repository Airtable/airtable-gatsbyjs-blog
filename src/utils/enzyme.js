// configuring enzyme to use the adapter for React 16
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

export default Enzyme.configure({ adapter: new Adapter() });
