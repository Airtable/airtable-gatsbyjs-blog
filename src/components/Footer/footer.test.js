import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Footer from './index';
import enzymeconf from '../../utils/enzyme';

it('renders correctly', () => {
  const tree = shallow(<Footer />);
  expect(toJson(tree)).toMatchSnapshot();
});
