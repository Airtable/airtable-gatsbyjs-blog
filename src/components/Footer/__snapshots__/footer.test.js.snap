// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-vt9sph"
>
  <Styled(div)
    align="left"
    color="white"
    p={
      Array [
        3,
        3,
        4,
      ]
    }
  >
    <Styled(div)
      justify="space-between"
      wrap={
        Array [
          "wrap",
          "wrap",
          "nowrap",
        ]
      }
    >
      <Styled(div)
        width={
          Array [
            1,
            1,
            0.3333333333333333,
          ]
        }
      >
        <Styled(div)>
          <div>
            <p>
              Gatbsythemes.com starter
            </p>
          </div>
        </Styled(div)>
      </Styled(div)>
      <Styled(div)
        width={
          Array [
            1,
            1,
            0.3333333333333333,
          ]
        }
      >
        <Styled(div)>
          <ul
            className="css-q5t0fb"
          >
            <li>
              <GatsbyLink
                to="/"
              >
                Home
              </GatsbyLink>
            </li>
            <li>
              <GatsbyLink
                to="/privacy"
              >
                Privacy Policy
              </GatsbyLink>
            </li>
            <li>
              <GatsbyLink
                to="/terms"
              >
                Terms of Service
              </GatsbyLink>
            </li>
          </ul>
        </Styled(div)>
      </Styled(div)>
      <Styled(div)
        width={
          Array [
            1,
            1,
            0.3333333333333333,
          ]
        }
      >
        <Styled(div)>
          <ul
            className="css-1jsczpv"
          >
            <li>
              <a
                href="https://facebook.com"
                rel="noopener"
                target="_blank"
                title="Link to our Facebook page"
              >
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"css-xvpfj7 feather feather-facebook\\"><path d=\\"M18 2h-3a5 5 0 0 0-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 0 1 1-1h3z\\"></path></svg>",
                    }
                  }
                />
              </a>
            </li>
            <li>
              <a
                href="https://twitter.com"
                rel="noopener"
                target="_blank"
                title="Link to our Twitter account"
              >
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"css-xvpfj7 feather feather-twitter\\"><path d=\\"M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z\\"></path></svg>",
                    }
                  }
                />
              </a>
            </li>
            <li>
              <a
                href="https://instagram"
                rel="noopener"
                target="_blank"
                title="Link to out Instragram account"
              >
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"css-xvpfj7 feather feather-instagram\\"><rect x=\\"2\\" y=\\"2\\" width=\\"20\\" height=\\"20\\" rx=\\"5\\" ry=\\"5\\"></rect><path d=\\"M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z\\"></path><line x1=\\"17.5\\" y1=\\"6.5\\" x2=\\"17.5\\" y2=\\"6.5\\"></line></svg>",
                    }
                  }
                />
              </a>
            </li>
            <li>
              <a
                href="mailto:"
                title="Our E-Mail address"
              >
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"css-xvpfj7 feather feather-mail\\"><path d=\\"M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z\\"></path><polyline points=\\"22,6 12,13 2,6\\"></polyline></svg>",
                    }
                  }
                />
              </a>
            </li>
          </ul>
        </Styled(div)>
      </Styled(div)>
    </Styled(div)>
  </Styled(div)>
  <Styled(div)
    align="center"
    px={
      Array [
        3,
        3,
        1,
      ]
    }
  >
    <p>
      Copyright Â© 2017 Gatbsythemes.com starter. All rights reserved.
    </p>
  </Styled(div)>
</div>
`;
